### PR TITLE
Add mint module integration testing 

### PR DESCRIPTION
**Pull request by BrandonWeng**
## Describe your changes and provide context
Adds mint integration tests

## Testing performed to validate your change
PR tests 

also ran locally:
image

### CODE REVIEW

File: integration_test/mint_module/mint_test.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      expr: START_DATE == LAST_MINT_DATE
+    - type: eval
+      expr: END_DATE > START_DATE
+    - type: eval
+      expr: TOTAL_MINT_AMOUNT == 999999999999
+    - type: eval
+      expr: REMAINING_MINT_AMOUNT == 666666666666
+    - type: eval
+      expr: LAST_MINT_AMOUNT == 333333333333
+    - type: eval
```
Comment (BrandonWeng) [inline integration_test/mint_module/mint_test.yaml:41]: let me also add another eval to check total supply

File: codecov.yml [hunk 2]
```diff
@@ -21,3 +21,5 @@ ignore:
   - "x/**/test_common.go"
   - "scripts/"
   - "contrib"
+  - "intetgration_test/**/*"
```
Comment (yzang2019) [inline codecov.yml:24]: tbh, I'm not sure if codcov is able to detect our customized integration test coverage for the chain

File: docker/localnode/scripts/step2_genesis.sh [hunk 3]
```diff
@@ -23,6 +23,9 @@ override_genesis '.app_state["staking"]["params"]["max_validators"]="50"'
 override_genesis '.consensus_params["block"]["max_gas"]="5000000000"'
 override_genesis '.app_state["staking"]["params"]["unbonding_time"]="10s"'
 
+# Set a token release schedule for the genesis file
+override_genesis "-arg start_date \"$(date +"%Y-%m-%d")\" --arg end_date \"$(date -d "+3 days" +"%Y-%m-%d")\" '.app_state[\"mint\"][\"params\"][\"token_release_schedule\"]=[{\"start_date\": \$start_date, \"end_date\": \$end_date, \"token_release_amount\": \"999999999999\"}]'"
```
Comment (yzang2019) [inline docker/localnode/scripts/step2_genesis.sh]: I wonder if we can split this into multiple simple override_genesis or not

File: integration_test/mint_module/mint_test.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+- name: Mint event triggered
+  inputs:
+    - cmd: seid q mint minter --output --json | jq -r ".denom"
```
Comment (yzang2019) [inline integration_test/mint_module/mint_test.yaml:6]: Can we add some comments for some critical steps about what the command is intended to do?

File: codecov.yml (hunk 2 above)
Comment (BrandonWeng) [reply codecov.yml:24]: I think overall it does seem to affect the coverage: 
image

File: integration_test/mint_module/mint_test.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      env: LAST_MINT_HEIGHT
+    # Compare the total supply and make sure that it increased
+    - cmd: seid q bank total --output json | jq -r '.supply[] | select(.denom=="usei") | .amount'
+      env: TOTAL_SUPPLY
+  verifiers:
+    - type: eval
+      expr: DENOM == "usei"
+    # Start date should be equal to last mint date since there's only one mint on the first day
+    - type: eval
+      expr: START_DATE == LAST_MINT_DATE
```
Comment (yzang2019) [inline integration_test/mint_module/mint_test.yaml:30]: Nit: you can do 
START_DATE == LAST_MINT_DATE and END_DATE > START_DATE 
to merge them together

**Review by yzang2019 — APPROVED** [review]

**Pull request merged**